Add system.cpu.physical.count and system.cpu.logical.count metrics (#99)
open-telemetry/semantic-conventions: docs/system/system-metrics.md

| ------------------------- | ---------------------------------------------------------------------------------------------------------------- | ----- | ------------------------------------------------- | ---------- | ---------------- | ----------------------------------- |
| system.cpu.time           | Seconds each logical CPU spent on each mode                                                                      | s     | Counter                                           | Double     | state            | idle, user, system, interrupt, etc. |
|                           |                                                                                                                  |       |                                                   |            | cpu              | Logical CPU number [0..n-1]         |
| system.cpu.utilization    | Difference in system.cpu.time since the last measurement, divided by the elapsed time and number of logical CPUs | 1     | Gauge                                             | Double     | state            | idle, user, system, interrupt, etc. |

> |                           |                                                                                                                  |       |                                                   |            | cpu              | Logical CPU number (0..n)           |
> | system.cpu.physical.count | Reports the number of actual physical processor cores on the hardware | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |
> | system.cpu.logical.count  | Reports the number of logical (virtual) processor cores created by the operating system to manage multitasking | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |

Reviewer: ```suggestion
|                           |                                                                                                                  |       |                                                   |            | cpu              | Logical CPU number (0..n)           |
| system.cpu.physical.count | Reports the number of actual physical processor cores on the hardware | {cpu} | UpDownCounter                                     | Int64      |                  |                                     |
| system.cpu.logical.count  | Reports the number of logical (virtual) processor cores created by the operating system to manage multitasking | {cpu} | UpDownCounter                                     | Int64      |                  |                                     |
```

(this would not be an issue after https://github.com/open-telemetry/semantic-conventions/pull/99#issuecomment-1660510502)

|                           |                                                                                                                  |       |                                                   |            | cpu              | Logical CPU number [0..n-1]         |
| system.cpu.utilization    | Difference in system.cpu.time since the last measurement, divided by the elapsed time and number of logical CPUs | 1     | Gauge                                             | Double     | state            | idle, user, system, interrupt, etc. |
|                           |                                                                                                                  |       |                                                   |            | cpu              | Logical CPU number (0..n)           |
| system.cpu.physical.count | Reports the number of actual physical processor cores on the hardware | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |

> | system.cpu.logical.count  | Reports the number of virtual processor cores created by the operating system to manage multitasking | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |

Reviewer 2: I would use `logical` term more as it's referred to in other metrics' descriptions.
```suggestion
| system.cpu.logical.count  | Reports the number of logical (virtual) processor cores created by the operating system to manage multitasking | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |
```

| system.cpu.time           | Seconds each logical CPU spent on each mode                                                                      | s     | Counter                                           | Double     | state            | idle, user, system, interrupt, etc. |
|                           |                                                                                                                  |       |                                                   |            | cpu              | Logical CPU number [0..n-1]         |
| system.cpu.utilization    | Difference in system.cpu.time since the last measurement, divided by the elapsed time and number of logical CPUs | 1     | Gauge                                             | Double     | state            | idle, user, system, interrupt, etc. |
|                           |                                                                                                                  |       |                                                   |            | cpu              | Logical CPU number (0..n)           |

> | system.cpu.physical.count | Reports the number of available physical CPUs - may change over time                                             | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |
> | system.cpu.logical.count  | Reports the number of available logical CPUs - may change over time                                              | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |

Reviewer 2: I was thinking about putting it into the description instead of a note. let me know WDYT
```suggestion
| system.cpu.physical.count | Reports the number of available physical CPUs (actual physical processor cores on the hardware)                                             | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |
| system.cpu.logical.count  | Reports the number of available logical CPUs (virtual processor cores created by the operating system to manage multitasking)                                              | {cpu} | UpDownCounters                                    | Int64      |                  |                                     |
```
